assets/js/components/Dashboard/PaMessagesPage.tsx

```diff
@@ -116,20 +119,12 @@ const PaMessagesPage: ComponentType = () => {
     });
   }, [setParams, stateFilter, serviceTypes]);
 
-  const { data, isLoading } = usePaMessages({ serviceTypes, stateFilter });
-  const shouldShowLoadingState = useDelayedLoadingState(isLoading);
-
-  const updatePaMessage = (updatedPaMessage: PaMessage) => {
-    if (!data) return;
+  const { data, isLoading } = usePaMessages({
+    serviceTypes,
+    stateFilter,
+  });
 
-    const updatedData = data.map((paMessage) => {
-      return paMessage.id === updatedPaMessage.id
-        ? updatedPaMessage
-        : paMessage;
-    });
-
-    mutate(`/api/pa-messages?${params.toString()}`, updatedData, false);
```

Comment:
I don't quite understand the implications of how you've changed the state tracking here. Are there any changes to how the UI behaves in practice?

Reply:
I'd say the key difference is ending the message or switching the toggle will now invalidate the client state and trigger a refetch. The UI behavior should be unchanged outside of maybe outside changes showing as a result of the refetch. I felt going the new direction simplified callbacks a bit.